game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/retreat/DefensiveSubsRetreatTest.java

```diff
@@ -118,41 +129,200 @@ void hasNoNamesWhenCanNotSubmergeAndNoRetreatTerritories() {
+          new DefensiveSubsRetreat(battleState, battleActions);
+
+      defensiveSubsRetreat.execute(executionStack, delegateBridge);
+
+      verify(battleState).retreatUnits(BattleState.Side.DEFENSE, retreatingUnits);
+    }
+
+    @Test
+    void retreatHappensWhenHasNoRetreatTerritoriesButDefendingIsSubmersible() {
+      final GameData gameData =
+          MockGameData.givenGameData()
+              .withSubmersibleSubs(false)
+              .withSubmarinesDefendingMaySubmergeOrRetreat(true)
+              .build();
+
+      final Unit unit = givenRealUnitCanEvade(gameData, defender);
+
+      final Collection<Unit> retreatingUnits = List.of(unit);
+      final BattleState battleState =
+          spy(
+              givenBattleStateBuilder()
+                  .battleSite(battleSite)
+                  .defender(defender)
+                  .defendingUnits(retreatingUnits)
+                  .gameData(gameData)
+                  .build());
+
+      when(battleActions.queryRetreatTerritory(
+              battleState,
+              delegateBridge,
+              defender,
+              List.of(battleSite),
+              true,
+              "defender retreat subs?"))
+          .thenReturn(battleSite);
+
+      final DefensiveSubsRetreat defensiveSubsRetreat =
+          new DefensiveSubsRetreat(battleState, battleActions);
+
+      defensiveSubsRetreat.execute(executionStack, delegateBridge);
+
+      verify(battleState).retreatUnits(BattleState.Side.DEFENSE, retreatingUnits);
+    }
   }
 
-  @Test
-  void retreatHappensWhenHasNoRetreatTerritoriesButIsSubmersible() {
-    final BattleState battleState =
-        givenBattleStateBuilder()
-            .defendingUnits(List.of(givenUnitCanEvade()))
-            .gameData(givenGameData().withSubmersibleSubs(true).build())
-            .build();
-    final DefensiveSubsRetreat defensiveSubsRetreat =
-        new DefensiveSubsRetreat(battleState, battleActions);
-
-    defensiveSubsRetreat.execute(executionStack, delegateBridge);
-
-    verify(battleActions)
-        .queryRetreat(eq(true), eq(MustFightBattle.RetreatType.SUBS), eq(delegateBridge), any());
+  @Nested
+  class RetreatHappens {
+
+    @Mock IDisplay display;
+    @Mock ISound sound;
+    @Mock IDelegateHistoryWriter historyWriter;
+    @Mock UnitCollection battleSiteCollection;
+
+    @BeforeEach
+    public void setupMocks() {
+      when(delegateBridge.getDisplayChannelBroadcaster()).thenReturn(display);
+      when(delegateBridge.getSoundChannelBroadcaster()).thenReturn(sound);
+      when(delegateBridge.getHistoryWriter()).thenReturn(historyWriter);
+      when(defender.getName()).thenReturn("defender");
+      when(battleSite.getUnitCollection()).thenReturn(battleSiteCollection);
+      when(battleSiteCollection.getHolder()).thenReturn(battleSite);
+    }
+
+    @Test
+    void retreatHappensWhenNotSubmersibleButHasRetreatTerritories() {
+      final GameData gameData = MockGameData.givenGameData().build();
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'MockGameData.givenGameData' due to existing static import 'games.strategy.triplea.delegate.battle.steps.MockGameData.givenGameData'](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6208307)